lib/Parser/RegexParser.cpp

```diff
@@ -2147,22 +2153,27 @@ namespace UnifiedRegex
             else if (nextChar == '\\')
             {
                 Node* returnedNode = ClassEscapePass1(&deferredCharNode, &deferredSetNode, previousWasASurrogate);
+                codePointToSet = pendingCodePoint;
 
                 if (returnedNode->tag == Node::MatchSet)
                 {
-                    codePointToSet = pendingCodePoint;
+                    if (unicodeFlagPresent && pendingRangeStart != INVALID_CODEPOINT)
```

Comment:
>pendingRangeStart != INVALID_CODEPOINT [](start = 46, length = 38)

This logic reads a bit confusingly. I'd combine the pendingRangeStart blocks and check `unicodeFlagPresent` inside. Move `pendingCodePoint = INVALID_CODEPOINT to the beginning of the block (after the check for FAIL)